globus_cli/services/transfer/async_transfer.py

```diff
@@ -55,18 +55,29 @@
         $ cat dat
         # file 1, simple
         ~/dir1/sourcepath1 /abspath/destpath1
+
         # file 2, with spaces in dest path
-        # paths without explicit ~ implicitly use ~
+        # paths without / or ~ are relative to the default_directory of the
+        # endpoint they are on
         dir2/sourcepath2   "path with spaces/dest2"
+
         # dir 1, requires --recursive option
         --recursive ~/srcdir1/ /somepath/destdir1
+
+        # if dest path is omitted, the source path is copied
+        # the lines below are therefore equivalent to
+        #   --recursive ~/srcdir1/ ~/srcdir1/
+        #   file_to_copy file_to_copy
+        --recursive ~/srcdir1/
+        file_to_copy
```

Comment:
Click is rewrapping these code block "paragraphs" and it looks a mess on the `--help` output. Extra `\b` escapes at every indentation "paragraph" seem to help.